services/pull/patch.go

```diff
@@ -41,9 +41,19 @@ func DownloadDiffOrPatch(ctx context.Context, pr *issues_model.PullRequest, w io
 	}
 	defer closer.Close()
 
-	if err := gitRepo.GetDiffOrPatch(pr.MergeBase, pr.GetGitRefName(), w, patch, binary); err != nil {
-		log.Error("Unable to get patch file from %s to %s in %s Error: %v", pr.MergeBase, pr.HeadBranch, pr.BaseRepo.FullName(), err)
-		return fmt.Errorf("Unable to get patch file from %s to %s in %s Error: %w", pr.MergeBase, pr.HeadBranch, pr.BaseRepo.FullName(), err)
+	compareString := pr.MergeBase + "..." + pr.GetGitRefName()
+	switch {
+	case patch:
+		err = gitRepo.GetPatch(compareString, w)
+	case binary:
+		err = gitRepo.GetDiffBinary(compareString, w)
+	default:
+		err = gitRepo.GetDiff(compareString, w)
+	}
```

Comment:
I still do not understand why you must introduce the `parseCompareArgs`.

Here you only uses "a...b", it could be just passed into GetPatch, etc, and use it as-is, without any parsing or trick.

Why you insist to pass `a...b`, parse it to `[]{"a...b"}`, then use `[]{"a...b"}` to add git arguments? Why not just use `a...b` directly and remove `parseCompareArgs`?

Reply:
removed all unnecessary code db99fae0d0560b46190a18ebe0ebed9fef95c325